x-pack/plugins/lens/public/visualizations/xy/xy_config_panel/annotations_config_panel/annotations_panel.tsx

```diff
@@ -132,13 +132,26 @@ export const AnnotationsPanel = (
               currentAnnotation?.type === 'query' ? 'query' : 'staticDate'
             }`}
             onChange={(id) => {
+              const type = id === `lens_xyChart_annotation_query` ? 'query' : 'manual';
+              const key =
+                !isQueryBased && id === `lens_xyChart_annotation_query`
+                  ? { type: 'point_in_time' }
+                  : currentAnnotation?.key;
+
+              const currentIndexPattern = frame.dataViews.indexPatterns[localLayer.indexPatternId];
               setAnnotations({
-                type: id === `lens_xyChart_annotation_query` ? 'query' : 'manual',
-                // when switching to query, reset the key value
-                key:
-                  !isQueryBased && id === `lens_xyChart_annotation_query`
-                    ? { type: 'point_in_time' }
-                    : currentAnnotation?.key,
+                type,
+                key,
+                ...(type === 'query'
+                  ? {
+                      timeField:
+                        currentIndexPattern.timeFieldName ??
```

Comment:
This doesn't work for two reasons:
* `timeFieldName` is an empty string if it's not set
* `timeField` in the annotation should be a string (the actual field name), but this code would set it to `IndexPatternField` instead.

Let's do the following:
* Type this function more strictly (right now there is no type safety here)
* Fix the problem and add a unit test